elasticgraph-warehouse/spec/unit/elastic_graph/warehouse/schema_definition/api_extension_spec.rb

```diff
@@ -0,0 +1,336 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse/schema_definition/api_extension"
+
+RSpec.describe ElasticGraph::Warehouse::SchemaDefinition::APIExtension, :unit do
+  include ElasticGraph::SchemaDefinition::TestSupport
+
+  it "extends the API factory with FactoryExtension" do
+    require "elastic_graph/warehouse/schema_definition/api_extension"
+
+    results = define_schema(schema_element_name_form: :snake_case, extension_modules: [ElasticGraph::Warehouse::SchemaDefinition::APIExtension]) do |s|
+      s.json_schema_version 1
+
+      # Just define a simple type to test the factory extension
+      s.object_type "Test" do |t|
+        t.field "id", "ID"
+      end
+    end
+
+    # The factory should have been extended during schema definition
+    expect(results).not_to be_nil
+  end
+
+  it "extends built-in scalar types with ScalarTypeExtension" do
+    require "elastic_graph/warehouse/schema_definition/api_extension"
+
```

Comment:
```suggestion
```

You shouldn't have to require this here as you've already required this at the top of this file.